File: covilo-core/src/main/java/me/kvdpxne/covilo/domain/model/CrimeClassification.java
package me.kvdpxne.covilo.domain.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

import java.util.LinkedHashSet;
import java.util.Set;
import java.util.UUID;

@Builder(toBuilder = true)
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Entity(name = "crime_classification")
@Table(name = "_crime_classification")
public class CrimeClassification {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  @Column(name = "_identifier", nullable = false)
  private UUID identifier;

  @Column(name = "_name", nullable = false, unique = true)
  private String name;

  @ToString.Exclude
  @ManyToMany(mappedBy = "classifications")
  private Set<Crime> crimes = new LinkedHashSet<>();

}

File: covilo-core/src/main/kotlin/me/kvdpxne/covilo/domain/model/CrimeClassification.kt
package me.kvdpxne.covilo.domain.model

import java.io.Serializable
import java.util.UUID

/**
 * Shortcut to the [CrimeClassification] collection.
 */
typealias CrimeClassifications = Collection<CrimeClassification>

/**
 * @param identifier
 * @param key
 */
data class CrimeClassification(
  val identifier: UUID = UUID.randomUUID(),
  var key: String
) : Serializable {

  companion object {
    private const val serialVersionUID: Long = 1639673346L
  }
}
